src/python/Utils/IteratorTools.py

```diff
@@ -50,3 +51,15 @@ def convertFromUnicodeToBytes(data):
         return type(data)(list(map(convertFromUnicodeToBytes, data)))
     else:
         return data
+
+
+def noDupListOfLists(listObj):
```

Comment:
I'm not sure that logic of this function cover all use cases, e.g. what if you have overlap slices like this:
```
[[1,10], [4,12]]
```
In fact, here is what python says about it:
```
>>> from itertools import islice, chain, groupby
>>> obj=[[1,10], [4,12]]
>>> obj.sort()
>>> list(k for k, _ in groupby(obj))
[[1, 10], [4, 12]]
```
Therefore, you either should change the docstring and describe the use-case correctly or change the logic of the function if we need to cover such use-cases.

I also suggest to change the function name from being to something like `removeDuplicateListRanges` which is what it is doing right now, and change its docstring accordingly to reflect its logic. It removes duplicate list ranges but it allows duplicate list entries.

Reply:
Valentin, perhaps I could rename it to something like `renameDuplicateListElements`?

I don't understand your first comment though. The use case is clear and it's described in the docstring (and covered by unit tests). I don't understand what "overlap" you mean with `[[1,10], [4,12]]`, please clarify.

The use case is simple, you have a list where each element is another list (list of lists). The goal of this function is to make the list of elements unique (so maybe saying remove duplicates isn't great either). E.g., if you have:
```
[[1,10], [4,12], [1,10]]
```
the output of this function should be (regardless of the order of the elements):
```
[[1,10], [4,12]]
```

Reply:
Maybe named it as `makeListElementsUnique` would be better?

Reply:
few things here:
- if you call them tuples then why do you use lists, the list of tuples would be `[(1,10), (4,12)]`
- if your lists represent range, e.g. `[1,10]` is short cut for `[1,2,3,4,5,6,7,8,9,10]`, then you have overlap in ranges

In either way I suggest to make proper clarification either in docstring or logic.

Said that, `makeListElementsUnique ` seems more appropriate but if you treat it as pairs, i.e. tuples, but in this case it is better to call them pairs or tuples and use `tuple` data types.